pkg/cvo/cvo.go

```diff
@@ -223,17 +227,32 @@ func New(
 // controller that loads and applies content to the cluster. It returns an error if the payload appears to
 // be in error rather than continuing.
 func (optr *Operator) InitializeFromPayload(restConfig *rest.Config, burstRestConfig *rest.Config) error {
-	update, err := payload.LoadUpdate(optr.defaultPayloadDir(), optr.releaseImage, optr.exclude)
+	update, err := payload.LoadUpdate(optr.defaultPayloadDir(), optr.release.Image, optr.exclude)
 	if err != nil {
 		return fmt.Errorf("the local release contents are invalid - no current version can be determined from disk: %v", err)
 	}
-	// XXX: set this to the cincinnati version in preference
-	if _, err := semver.Parse(update.ImageRef.Name); err != nil {
-		return fmt.Errorf("the local release contents name %q is not a valid semantic version - no current version will be reported: %v", update.ImageRef.Name, err)
+	optr.release.Version = update.Metadata.Version
```

Comment:
Does `update.Metadata.Version` = `update.ReleaseVersion`? Wondering why they are two and diff if any.

Reply:
Yes, [they match][1], or we fail to load the release.

[1]: https://github.com/openshift/cluster-version-operator/commit/f666975b299ba42b9d2194ecbcac2e6cf62ce2f8#diff-8fadc5d8bd084451d026d5bf128dad66R263-R269